### PR TITLE
Attempt to determine download URL from dmg metadata.

### LINKED ISSUE #81: Get download url from com.apple.metadata:kMDItemWhereFroms if it exists
When analyzing a dmg, it's possible that the `com.apple.metadata:kMDItemWhereFroms` metadata could reveal a download URL.

### PR DESCRIPTION
**Pull request by homebysix**
Closes #81.

### CODE REVIEW

File: scripts/recipe_robot_lib/inspect.py [hunk 1]
```diff
@@ -614,6 +614,29 @@ def inspect_disk_image(input_path, args, facts):
     else:
         facts["inspections"].append("disk_image")
 
+    # See if we can determine the download URL from the file metadata.
+    if "download_url" not in facts:
+        cmd = "/usr/bin/xattr -p com.apple.metadata:kMDItemWhereFroms \"%s\"" % input_path
```
Comment (gregneagle) [inline scripts/recipe_robot_lib/inspect.py]: python has an matter module; no need to call out to xattr (which if you look, you will discover xattr _is_ a Python script!)

File: scripts/recipe_robot_lib/inspect.py [hunk 2]
```diff
@@ -614,6 +614,29 @@ def inspect_disk_image(input_path, args, facts):
...
+        exitcode, out, err = get_exitcode_stdout_stderr(cmd)
+        if exitcode == 0:
+            # Save the hex output as a file.
+            where_froms_path = os.path.join(CACHE_DIR, "kMDItemWhereFroms.plist")
+            with open(where_froms_path, "wb") as where_froms_file:
+                where_froms_file.write(out)
+            # Convert the hex to a binary plist.
```
Comment (gregneagle) [inline scripts/recipe_robot_lib/inspect.py]: Using the Python xattr module would also eliminate the need to convert to binary plist -- the value returned is already the bplist

File: scripts/recipe_robot_lib/inspect.py [hunk 3]
```diff
@@ -614,6 +614,29 @@ def inspect_disk_image(input_path, args, facts):
...
+            cmd = "/usr/bin/xxd -r -p -g 1 -c 16 \"%s\" -" % where_froms_path
+            exitcode, out, err = get_exitcode_stdout_stderr(cmd)
+            with open(where_froms_path, "wb") as where_froms_file:
+                where_froms_file.write(out)
+            # Convert the binary plist to xml1 plain text.
```
Comment (gregneagle) [inline scripts/recipe_robot_lib/inspect.py]: If you are using FoundationPlist there is no reason to convert to XML!

Comment (gregneagle) [inline scripts/recipe_robot_lib/inspect.py]: Pulling it all together:

```python
import FoundationPlist
import xattr

foo = xattr.getxattr("/Users/gneagle/Downloads/munkitools-latest.dmg", "com.apple.metadata:kMDItemWhereFroms")
bar = FoundationPlist.readPlistFromString(foo)

print bar
print bar[0]
```




```

Comment (homebysix) [inline scripts/recipe_robot_lib/inspect.py]: Much much simpler. Thank you!

**Pull request merged**